### PR TITLE
Add pagination to the output of the `favorites list` command

### LINKED ISSUE #606: Problem - Favorites
What if there are to many favourites to list with /favorites list? As i see there is no page system for that one. So if it would be to many the bot CANT list it

### PR DESCRIPTION
**Pull request by coryjreid**
Closes #606

Introduces pagination to the output of the `favorites list` command to prevent extremely large lists of favorites violating Discord's embed/message limits.

- [X] I updated the changelog

Key Choices
- **Opted for a pagination library rather than reinventing the wheel**
   I figured it would be easier to maintain by simply adding a package which provides the pagination rather than creating one myself.
- **Updated Discord.js Due to Encountered Bug**
   I was encountering a bug when using the pagination library - turns out the bug was in Discord.js and upgrading resolved the issue. I updated and pinned to latest. (See: https://stackoverflow.com/questions/75892478/discord-js-v14-7-1-typeerror-channel-istextbased-is-not-a-function)

Note: Edits to the changelog might seem excessive... I simply formatted the file after making my additions to bring solid consistency. Can always undo this if this is an issue.

### CODE REVIEW

**Comment by coryjreid** [comment]
Alright, I got all the build checks passing! Sorry for the spam... I'm a Java developer typically so not super familiar with the Node/TS world.

However it shows the merge is still blocked due to my first commits being unsigned (had never had a need to have that setup before).

I'm not sure if you can simply override that on a case-by-case basis or if I should close this PR and open a new one?

Please do let me know at your leisure (also if there's anything else I need to do/adjust)! Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,296 +1,477 @@
...
+
+### Fixed
+
+- Favorites list exceeding Discord's size limit could not be
+  viewed ([#606](https://github.com/codetheweb/muse/issues/606))
+
 ## [2.3.0] - 2023-05-13
+
 ### Added
+
```
Comment (codetheweb) [inline CHANGELOG.md:24]: I should get this file auto formatted, but let's revert the formatting changes for now

**Review by codetheweb — APPROVED** [review]
tried it out and it looks awesome!

File: tsconfig.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "compilerOptions": {
-    "lib": ["esnext"],
-    "target": "es2018",
-    "module": "ES2020",
+    "lib": ["es2020" ,"esnext"],
+    "target": "es2020",
+    "module": "es2020",
```
[the remark this answers is not in the document]
Comment (coryjreid) [reply tsconfig.json:5]: The `yarn tsc` command was failing due to issues with project configuration and conflicting versions of `@types/ws`. This was cropping up in one of the Workflows.

[the remark this answers is not in the document]
Comment (codetheweb) [reply tsconfig.json:5]: hmm, I reverted this change and it still worked--could you try again and see if it's not working for you?

[the remark this answers is not in the document]
Comment (coryjreid) [reply tsconfig.json:5]: Sure. So this issue was cropping up in the GitHub checks, not locally. Though I was able to run the commands locally and reproduce the issue.

Now the error shows up in a different dependency from when I first saw it. It was causing me not to get all green checkmarks on the tasks on GitHub so I worked on finding a solution (and my solution was to bump the targets as you saw).

I used WebStorm, Node JS v18.17.0 and yarn 1.22.19. I reproduced this on both MacOS and Windows.

image

```
{
  "compilerOptions": {
    "lib": ["esnext"],
    "target": "es2018",
    "module": "ES2020",
    "moduleResolution": "node",
    "strict": true,
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "sourceMap": true,
    "resolveJsonModule": true,
    "noEmit": true
  },
  "include": ["src"],
  "exclude": ["node_modules"]
}
```

```
PS D:\Projects\muse> yarn tsc
yarn run v1.22.19
$ D:\Projects\muse\node_modules\.bin\tsc
node_modules/@sapphire/snowflake/dist/index.d.ts:8:33 - error TS2737: BigInt literals are not available when targeting lower than ES2020.

8 declare const MaximumWorkerId = 31n;
                                  ~~~

node_modules/@sapphire/snowflake/dist/index.d.ts:12:34 - error TS2737: BigInt literals are not available when targeting lower than ES2020.

12 declare const MaximumProcessId = 31n;
                                    ~~~

node_modules/@sapphire/snowflake/dist/index.d.ts:16:34 - error TS2737: BigInt literals are not available when targeting lower than ES2020.

16 declare const MaximumIncrement = 4095n;
                                    ~~~~~


Found 3 errors in the same file, starting at: node_modules/@sapphire/snowflake/dist/index.d.ts:8

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
PS D:\Projects\muse>
```

[the remark this answers is not in the document]
Comment (codetheweb) [reply tsconfig.json:5]: ah, I see

looks like just the target needs to be updated 👍

**Pull request merged**